scheduler.go

```diff
@@ -1377,7 +1377,7 @@ func (s *Scheduler) CustomTimer(customTimer func(d time.Duration, f func()) *tim
 func (s *Scheduler) StopBlockingChan() {
 	s.startBlockingStopChanMutex.Lock()
 	if s.startBlockingStopChan != nil {
-		s.startBlockingStopChan <- struct{}{}
+		close(s.startBlockingStopChan)
```

Comment:
this can result in a panic if the method is called twice.

If you modify the stop method to look like:
```golang
func (s *Scheduler) stop() {
	s.stopJobs(s.jobs)
	s.executor.stop()
	s.StopBlockingChan()
	s.setRunning(false)
}
```
and this method to look like 

```golang
func (s *Scheduler) StopBlockingChan() {
	s.startBlockingStopChanMutex.Lock()
	if s.IsRunning() && s.startBlockingStopChan != nil {
		close(s.startBlockingStopChan)
	}
	s.startBlockingStopChanMutex.Unlock()
}
```
i believe that solves it

Reply:
I also added setting the channel to nil after closing so that if the method is called when someone does start async it won’t also panic on a closed channel

Reply:
Hm, looks like that resulted in a data race in the tests. I’ll look later if you’re aren’t able to get to it.

Reply:
@rfyiamcool this is the change I was looking to see for the StartBlocking method

```diff
func (s *Scheduler) StartBlocking() {
	s.StartAsync()
	s.startBlockingStopChanMutex.Lock()
	s.startBlockingStopChan = make(chan struct{}, 1)
	s.startBlockingStopChanMutex.Unlock()
	<-s.startBlockingStopChan
+	s.startBlockingStopChanMutex.Lock()
+	s.startBlockingStopChan = nil
+	s.startBlockingStopChanMutex.Unlock()
}
```